components/printer/loading.go

```diff
@@ -0,0 +1,27 @@
+package printer
+
+import (
+	"time"
+
+	"github.com/briandowns/spinner"
+)
+
+type Func[T any] func() (T, error)
```

Comment:
do you intent to use this type elsewhere? If not, I'd rather not export it.

Reply:
It can be private indeed. I just couldn't leave it out.